ownCloud/Client/Actions/EditDocumentViewController.swift

```diff
@@ -0,0 +1,168 @@
+ *
+ */
+
+import UIKit
+import ownCloudSDK
+import QuickLook
+
+@available(iOS 13.0, *)
+protocol EditDocumentViewControllerDelegate: class {
+	func editDocumentViewControllerDidDismiss(_ controller: EditDocumentViewController)
+}
+
+@available(iOS 13.0, *)
+class EditDocumentViewController: QLPreviewController, Themeable {
+
+	weak var core: OCCore?
+	var item: OCItem
+	var editDelegte: EditDocumentViewControllerDelegate?
+	var handleSaving: QLPreviewItemEditingMode?
+
+	var source: URL {
+		didSet {
+			if self.source != oldValue {
+				OnMainThread {
+					self.reloadData()
+				}
+			}
+		}
+	}
+
+	init(with file: URL, item: OCItem, core: OCCore? = nil) {
+		self.source = file
+		self.core = core
+		self.item = item
+
+		super.init(nibName: nil, bundle: nil)
+
+		self.dataSource = self
+		self.delegate = self
+		self.navigationItem.leftBarButtonItem = UIBarButtonItem(barButtonSystemItem: .done, target: self, action: #selector(dismissAnimated))
+
+		Theme.shared.register(client: self, applyImmediately: true)
+	}
+
+	@objc func dismissAnimated() {
+		self.setEditing(false, animated: false)
+
+		if handleSaving == nil {
+			requestHandleSaving { (_) in
+				self.dismiss(animated: true, completion: nil)
+			}
+		} else {
+			self.dismiss(animated: true, completion: nil)
+		}
+	}
+
+	func requestHandleSaving(completion: ((QLPreviewItemEditingMode) -> Void)? = nil) {
+		var presentationStyle: UIAlertController.Style = .actionSheet
+		if UIDevice.current.isIpad() {
+			presentationStyle = .alert
+		}
+
+		let alertController = ThemedAlertController(title: "How should this file be saved?".localized,
+													message: nil,
+													preferredStyle: presentationStyle)
+		alertController.addAction(UIAlertAction(title: "Cancel".localized, style: .cancel, handler: nil))
+
+		alertController.addAction(UIAlertAction(title: "Overwrite originale".localized, style: .default, handler: { (_) in
+			self.handleSaving = .updateContents
+
+			completion?(.updateContents)
+		}))
+
+		alertController.addAction(UIAlertAction(title: "Save as copy".localized, style: .default, handler: { (_) in
+			self.handleSaving = .createCopy
+
+			completion?(.createCopy)
+
+		}))
+
+		alertController.addAction(UIAlertAction(title: "Discard changes".localized, style: .destructive, handler: { (_) in
+			self.handleSaving = .disabled
+
+			completion?(.disabled)
+		}))
+
+		self.present(alertController, animated: true, completion: nil)
+	}
+
+	func saveModifiedContents(at url: URL, handleSaving: QLPreviewItemEditingMode) {
+		switch handleSaving {
+		case .createCopy:
+			if let core = core, let parentItem = item.parentItem(from: core) {
+				self.core?.importFileNamed(item.name, at: parentItem, from: url, isSecurityScoped: false, options: [ .automaticConflictResolutionNameStyle : OCCoreDuplicateNameStyle.bracketed.rawValue ], placeholderCompletionHandler: nil, resultHandler: nil)
```

Comment:
@hosy Are completion handlers of `OCCore` methods not used intentionally, because core would anyway create an issue which is going to be displayed to the user? May be at least adding some logging would make sense?